addons/website_event_sale/tests/test_frontend_buy_tickets.py

```diff
@@ -6,14 +6,23 @@
 from datetime import timedelta
 
 from odoo.addons.base.tests.common import HttpCaseWithUserDemo
+from odoo.addons.website_event_sale.tests.common import TestWebsiteEventSaleCommon
 from odoo.fields import Datetime
 
 
 @odoo.tests.common.tagged('post_install', '-at_install')
-class TestUi(HttpCaseWithUserDemo):
+class TestUi(HttpCaseWithUserDemo, TestWebsiteEventSaleCommon):
 
     def setUp(self):
         super().setUp()
+
+        self.rate = self.env['res.currency.rate'].create({
+            'company_id': self.env.company.id,
+            'currency_id': self.currency_test.id,
+            'name': '2022-01-01',
+        })
+        self.rate['company_rate'] = 10  # only way I found to override rate
```

Comment:
I don't know if it's much prettier, but we often see in the tests a clearing of demo data with `self.env['res.currency.rate'].search([]).unlink()`, so something like this should work:

```suggestion
        self.env['res.currency.rate'].search([]).unlink()
        self.rate = self.env['res.currency.rate'].create({
            'company_id': self.env.company.id,
            'currency_id': self.currency_test.id,
            'rate': 10.0,
        })
```
Would it be it possible to set this up in `TestWebsiteEventSaleCommon` with the currency definition? You wouldn't have to define it twice.

Reply:
Thanks, I was looking for something like that!